internal/storage/reload.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021-2022 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/singleflight"
+)
+
+var sfGroup singleflight.Group
+
+func Reload(ctx context.Context, log *zap.SugaredLogger, rs ReloadableStore) error {
```

Comment:
We should be using the logger from context because then the logging calls are request-scoped -- which is useful to identify the sequence of events that led to something. Also, use `*zap.Logger` instead of `*zap.SugaredLogger` where possible because it's more efficient.